### PR TITLE
Add UserMetricsModel to exception list for missing readOnly attribute.

### PR DESCRIPTION
**Pull request by ttlcgl**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/trafficmanager/resource-manager/readme.md [hunk 1]
```diff
@@ -51,6 +51,10 @@ directive:
     reason: This will cause breaking changes in .NET SDK
     from: trafficmanager.json
     where: $.definitions.Endpoint
+  - suppress: RequiredPropertiesMissingInResourceModel
+    reason: This will cause breaking changes in .NET SDK
+    from: trafficmanager.json
+    where: $.definitions.UserMetricsModel
```
Comment (anuchandy) [inline specification/trafficmanager/resource-manager/readme.md]: @tatlicioglu - this is to suppress the following ARMViolation I guess:

```
RequiredPropertiesMissingInResourceModel/R2020/ARMViolation: Model definition 'UserMetricsModel' must have the properties 'name', 'id' and 'type' in its hierarchy and these properties must be marked as readonly.
```

Looking at the definition of UserMetricsModel it does have `name, id and type` in the hierarchy (UserMetricsModel -> ProxyResource -> [Resource](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/trafficmanager/resource-manager/Microsoft.Network/stable/2018-04-01/trafficmanager.json)), but `id` is not readonly. 

As mentioned in the suppression reason, adding readonly attribute for Resource::id will cause breaking change in the SDK (e.g. [ctr](https://github.com/Azure/azure-sdk-for-net/blob/682e9583d4a69a275fcbab76278807a80b20418d/src/SDKs/TrafficManager/Management.TrafficManager/Generated/Models/UserMetricsModel.cs#L42) for UserMetricsModel will break) that requires a major version bump for the SDK. A major version upgrade of SDK for such a small change doesn't worth.

So we might want add the work of `marking id as readonly` as a TODO in major version upgrade work items. At this point i'm not sure when that happens. Maybe we should introduce a process of cleaning the swaggers once in a semester or so where we fix all these issues and do a major version upgrade.  Something that needs to be discussed within the team.

It looks like there are bunch of other models (Endpoint, Profile, TrafficManagerGeographicHierarchy & HeatMapModel) inherits from Resource has `RequiredPropertiesMissingInResourceModel`. Any reason not to suppress violations for those models?

\\cc @johanste, @salameer

**Review by ryansbenson — APPROVED** [review]
Reviewed

**Review by dsgouda — APPROVED** [review]

**Comment by anuchandy** [comment]
@dsgouda looks like we merged this, but my question on models is not addressed. What's exactly this PR solves by suppress ing violation on one model when there exits other models with similar violation?

**Comment by dsgouda** [comment]
@anuchandy We discussed these concerns with @tatlicioglu and @RyanBensonMSFT and decided to move forward with this change as a hotfix. Moving forward the TrafficManager RP will be modelling these models in a way that it doesn't violate rules around Resources

**Pull request merged**